Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp

```diff
@@ -0,0 +1,162 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/CalculatePlaczekSelfScattering.h"
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidDataObjects/WorkspaceCreation.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidKernel/Atom.h"
+#include "MantidKernel/Material.h"
+
+#include <utility>
+
+namespace Mantid {
+namespace Algorithms {
+
+DECLARE_ALGORITHM(CalculatePlaczekSelfScattering)
+
+void CalculatePlaczekSelfScattering::init() {
+  declareProperty(
+      std::make_unique<API::WorkspaceProperty<API::MatrixWorkspace>>(
+          "InputWorkspace", "", Kernel::Direction::Input),
+      "Workspace of fitted incident spectrum with it's first derivative. "
+      "Workspace must have instument and sample data.");
+  declareProperty(
+      std::make_unique<API::WorkspaceProperty<API::MatrixWorkspace>>(
+          "OutputWorkspace", "", Kernel::Direction::Output),
+      "Workspace with the Self scattering correction");
+}
+//----------------------------------------------------------------------------------------------
+/** Validate inputs.
+ */
+std::map<std::string, std::string>
+CalculatePlaczekSelfScattering::validateInputs() {
+  std::map<std::string, std::string> issues;
+  const API::MatrixWorkspace_sptr inWS = getProperty("InputWorkspace");
+  const Geometry::DetectorInfo det_info = inWS->detectorInfo();
+  if (det_info.size() == 0) {
+    issues["InputWorkspace"] =
+        "Input workspace does not have detector information";
+  }
+  Kernel::Material::ChemicalFormula formula =
+      inWS->sample().getMaterial().chemicalFormula();
+  if (formula.size() == 0) {
+    issues["InputWorkspace"] = "Input workspace does not have a valid sample";
+  }
+  return issues;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void CalculatePlaczekSelfScattering::exec() {
+  const API::MatrixWorkspace_sptr inWS = getProperty("InputWorkspace");
+  API::MatrixWorkspace_sptr outWS = getProperty("OutputWorkspace");
+  const double factor = 1.0 / 1.66053906660e-27;
```

Comment:
How about we add a comment line to note this is the value for `scipy.constants.physical_constants['atomic mass unit-kilogram relationship']`?

I forgot myself where this came from when I looked at it.

Reply:
Comment added, thanks!